src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala

```diff
@@ -72,78 +76,97 @@ trait NihService extends LazyLogging {
   val thurloeDao: ThurloeDAO
   val googleDao: GoogleServicesDAO
 
+  val nihWhitelists: Set[NihWhitelist] = FireCloudConfig.Nih.whitelists
+
   def getNihStatus(userInfo: UserInfo): Future[PerRequestMessage] = {
     thurloeDao.getProfile(userInfo) flatMap {
       case Some(profile) =>
         profile.linkedNihUsername match {
           case Some(_) =>
-            rawlsDao.isDbGapAuthorized(userInfo) map { isDbGapAuthorized =>
-              RequestComplete(NihStatus(profile.copy(linkExpireTime = Option(profile.linkExpireTime.getOrElse(0L))), Some(isDbGapAuthorized)))
+            Future.traverse(nihWhitelists) { whitelistDef =>
+              rawlsDao.isGroupMember(userInfo, whitelistDef.groupToSync).map(isMember => NihWhitelistStatus(whitelistDef.name, isMember))
+            }.map { whitelistMembership =>
+              RequestComplete(NihStatus(profile, whitelistMembership))
             }
           case None => Future.successful(RequestComplete(NotFound))
         }
       case None => Future.successful(RequestComplete(NotFound))
     }
   }
 
-  def updateNihLinkAndSyncSelf(userInfo: UserInfo, nihLink: NIHLink): Future[PerRequestMessage] = {
-    val syncWhiteListResult = syncWhitelistUser(userInfo.getUniqueId, nihLink.linkedNihUsername)
-    val profilePropertyMap = nihLink.propertyValueMap
-    val propertyUpdates = thurloeDao.saveKeyValues(userInfo, profilePropertyMap)
-
-    val profileResponse = propertyUpdates.map { response =>
-      if(response.isSuccess) RequestComplete(OK)
-      else RequestCompleteWithErrorReport(InternalServerError, "Error updating NIH link.")
+  private def downloadNihWhitelist(whitelist: NihWhitelist): Set[String] = {
+    val usersList = try {
+      Source.fromInputStream(googleDao.getBucketObjectAsInputStream(FireCloudConfig.Nih.whitelistBucket, whitelist.fileName))
+    } catch {
+      case ioe: IOException => {
+        logger.error(s"Unable to download ${whitelist.name} whitelist from Google Bucket (${FireCloudConfig.Nih.whitelistBucket}/${whitelist.fileName})")
+        //In the rare case where we cannot read the whitelist from the bucket, this will act as a fail-safe and
+        //act as if the whitelist is empty
+        Source.fromInputStream(new ByteArrayInputStream("".getBytes("UTF-8")))
```

Comment:
I now ask: is this the right behavior? It seems aggressive but maybe that's the right approach given what we're dealing with